batch/build-batch-worker-image-startup.sh

```diff
@@ -1,11 +1,16 @@
 #!/bin/bash
 
+curl -sSO https://dl.google.com/cloudagents/add-logging-agent-repo.sh
```

Comment:
I like to use the full flags when we can, and I think we should add `--fail`:

```
curl --silent --show-error --remote-name --fail \
     https://dl.google.com/cloudagents/add-logging-agent-repo.sh
```

```
       -f, --fail
              (HTTP) Fail silently (no output at all) on server errors. This is
              mostly done to better enable scripts etc to better deal with
              failed attempts. In normal cases when an HTTP server fails to
              deliver a document, it returns an HTML document stating so (which
              often also describes why and more). This flag will prevent curl
              from outputting that and return error 22.

              This method is not fail-safe and there are occasions where
              non-successful response codes will slip through, especially when
              authentication is involved (response codes 401 and 407).
```

Reply:
`--fail` with `--show-error` is a great combo that prints useful information but also ensures that we get an error code when something goes wrong. On that note, should this shell script have `set -ex`?